aptos-move/framework/aptos-framework/sources/TokenV1.move

```diff
@@ -0,0 +1,935 @@
+/// This module provides the foundation for Tokens.
+module AptosFramework::TokenV1 {
```

Comment:
why aren't we just putting this on top of Token?

Reply:
tried. There are a lot of dependencies on Token now, eg: transaction builder, smoke-test, token indexer, etc.  Then this larger PR will be much larger. 
I think it is better to iterate in a new module gradually and then replace the old Token and all the dependencies.